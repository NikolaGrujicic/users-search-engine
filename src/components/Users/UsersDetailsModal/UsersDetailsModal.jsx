import React from 'react';

const UsersDetailsModal = ({ state, city, street, postcode, phone, cell }) => (
  <div className="modal fade" id="basicModal" tabIndex="-1" role="dialog" aria-labelledby="basicModal" aria-hidden="true">
    <div className="modal-dialog">
      <div className="modal-content">
        <div className="modal-header">
          <h4 className="modal-title" id="myModalLabel">User Details</h4>
          <button type="button" className="close" data-dismiss="modal" aria-label="Close">
            <span aria-hidden="true">&times;</span>
          </button>
        </div>
        <div className="modal-body">
          <p>
            State:
            <u>{state}</u>
          </p>
          <p>
            City:
            <u>{city}</u>
          </p>
          <p>
            Street:
            <u>{street}</u>
          </p>
          <p>
            Postcode:
            <u>{postcode}</u>
          </p>
          <p>
            Phone:
            <u>{phone}</u>
          </p>
          <p>
            Cell:
            <u>{cell}</u>
          </p>
        </div>
        <div className="modal-footer">
          <button type="button" className="btn btn-default" data-dismiss="modal">Close</button>
        </div>
      </div>
    </div>
  </div>
);

export default UsersDetailsModal;
