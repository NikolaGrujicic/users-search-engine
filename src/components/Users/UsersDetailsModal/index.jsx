import UsersDetailsModal from './UsersDetailsModal';

export default UsersDetailsModal;
