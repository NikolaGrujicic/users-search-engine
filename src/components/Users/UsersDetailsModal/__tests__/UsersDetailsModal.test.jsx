import React from 'react';
import { shallow, mount } from 'enzyme';
import UsersDetailsModal from '../UsersDetailsModal';

test('renders Users Details Modal', () => {
  const wrapper = shallow(
    <UsersDetailsModal />,
  );
  expect(wrapper).toMatchSnapshot();
});

test('renders Users Details Modal with props', () => {
  const wrapper = shallow(
    <UsersDetailsModal
      state="California"
      city="Los Angeles"
      postcode="123456"
      street="Queens Boulevard"
      phone="555-333"
      cell="333-555"
    />,
  );

  expect(wrapper).toMatchSnapshot();
});

test('check Users Details Modal props are correct', () => {
  const wrapper = mount(
    <UsersDetailsModal
      state="California"
      city="Los Angeles"
      postcode="123456"
      street="Queens Boulevard"
      phone="555-333"
      cell="333-555"
    />,
  );

  expect((wrapper).prop('state')).toEqual('California');
  expect((wrapper).prop('city')).toEqual('Los Angeles');
  expect((wrapper).prop('postcode')).toEqual('123456');
  expect((wrapper).prop('street')).toEqual('Queens Boulevard');
  expect((wrapper).prop('phone')).toEqual('555-333');
  expect((wrapper).prop('cell')).toEqual('333-555');
});
