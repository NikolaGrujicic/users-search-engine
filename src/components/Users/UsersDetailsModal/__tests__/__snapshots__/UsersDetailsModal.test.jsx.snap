// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Users Details Modal 1`] = `
<div
  aria-hidden="true"
  aria-labelledby="basicModal"
  className="modal fade"
  id="basicModal"
  role="dialog"
  tabIndex="-1"
>
  <div
    className="modal-dialog"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <h4
          className="modal-title"
          id="myModalLabel"
        >
          User Details
        </h4>
        <button
          aria-label="Close"
          className="close"
          data-dismiss="modal"
          type="button"
        >
          <span
            aria-hidden="true"
          >
            Ã—
          </span>
        </button>
      </div>
      <div
        className="modal-body"
      >
        <p>
          State:
          <u />
        </p>
        <p>
          City:
          <u />
        </p>
        <p>
          Street:
          <u />
        </p>
        <p>
          Postcode:
          <u />
        </p>
        <p>
          Phone:
          <u />
        </p>
        <p>
          Cell:
          <u />
        </p>
      </div>
      <div
        className="modal-footer"
      >
        <button
          className="btn btn-default"
          data-dismiss="modal"
          type="button"
        >
          Close
        </button>
      </div>
    </div>
  </div>
</div>
`;
